### DISCUSSION TITLE
[Bug] Drag and drop stopped working after transferring from Windows to Debian

### DISCUSSION THREAD

**Discussion by sgjohnson1981** [piroor/treestyletab#3616] [Q&A]
## Steps to reproduce

I thought at first it had to do with my settings, which I imported from Windows, but even on a clean Firefox profile it doesn't work. When I attempt to drag and drop a tab in an "auto-hidden" sidebar the sidebar disappears behind the tab outline. If there is enough "hidden" sidebar to still see the tabs I can manage to nest the tab as long as it's a higher level tab. If it's nested further in the tree it's impossible to see. The sidebar won't reappear until I drop the tab and rehover. This worked fine on Windows. I had a userchrome.css that was slightly different than the one in the wiki, which worked in Windows, but the wiki one doesn't work either. I entered both the userchrome.css contents into a Browser Toolbox window and the extra styles into the TST settings. It didn't work.

## Expected result

Should be able to drag and drop.

## Actual result

See above

## Environment

 * Platform (OS): Debian 12
 * Version of Firefox:  115.14.0esr
 * Version (or revision) of Tree Style Tab:  4.0.21

**Comment by piroor** [comment]
It looks not a bug of TST but an issue of customization with userChrome.css. It is out of scope of this issue tracker.

**Reply by piroor** [reply]
I've moved this from an issue to a discussion. I hope someone find the way to solve the problem by writing something CSS customization.

**Reply by sgjohnson1981** [reply]
But I used the userchrome.css stuff from the wiki. Why would it not work between different versions of Firefox?

Here's my userchrome.css for anyone's reference.

/* Shrink sidebar until hovered, when using Tree Style Tab. */
:root {
    --thin-tab-width: 3px;
    --wide-tab-width: 500px;
}
#sidebar-box:not([sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"]) {
    min-width: var(--wide-tab-width) !important;
    max-width: none !important;
}
#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"] {
    position: relative !important;
    transition: all 100ms !important;
    min-width: var(--thin-tab-width) !important;
    max-width: var(--thin-tab-width) !important;
}
#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"]:hover {
    transition: all 200ms !important;
    min-width: var(--wide-tab-width) !important;
    max-width: var(--wide-tab-width) !important;
    margin-right: calc((var(--wide-tab-width) - var(--thin-tab-width)) * -1) !important;
    z-index: 1;

**Reply by piroor** [reply]
Firefox is a multi-platform application but many behaviors are affected on restrictions of each platform, for example, click-through (receiving a mouse event on a tab element while the window is behind another window) cannot happen on macOS. I never confirmed yet but I think that differences around `:hover` pseudo class may happen between Windows and Linux.
